internal/manager/setup.go

```diff
@@ -52,23 +72,23 @@ func setupControllerOptions(logger logr.Logger, c *Config, scheme *runtime.Schem
 		LeaderElectionID:       c.LeaderElectionID,
 		SyncPeriod:             &c.SyncPeriod,
 	}
+
 	// determine how to configure namespace watchers
-	switch len(c.WatchNamespaces) {
-	case 0:
-		// watch all namespaces
+	if len(c.WatchNamespaces) == 0 {
```

Comment:
Could we leave the logic as simple as it was, and use a concatenation of WatchNamespaces and the additional namespaces altogether as the input list of namespaces?

Reply:
No because it would change the behavior: we need to specifically know if `c.WatchNamespaces` was zero because that is our indication that we're going to watch ALL namespaces, which supports the Gateway caching needs anyway. If we did any appending or joining of the two for identification we would end up with a broken "all" mode.